### PR TITLE
Updates to bind mount for maximal event propagation enjoyment

### PR DESCRIPTION
**Pull request by rheinwein**
I opened this PR so it's a bit easier to see the diff.

### CODE REVIEW

File: docker_swarm/run.sh [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+docker build -t inotifywait .
+
 docker service create \
   --name docker_swarm \
-  --mount "type=bind,source=$(pwd)/watch,target=/opt/watch" \
-  timsabat/inotifywait
-
-for i in {1..5}; do echo "sleeping $i of 5"; sleep 1; done
+  --mount type=bind,source=$(pwd)/watch,target=/opt/watch,bind-propagation="" \
+  inotifywait
```
Comment (rheinwein) [inline docker_swarm/run.sh:8]: this is the special sauce right here-- `bind-propagation=""`

**Comment by rheinwein** [comment]
```
docker_swarm $ ./run.sh 
Sending build context to Docker daemon  5.12 kB
Step 1 : FROM node:5-slim
 ---> 4b56550d9eec
Step 2 : RUN apt-get update && apt-get install -y inotify-tools
 ---> Using cache
 ---> f59a3a185f31
Step 3 : RUN mkdir -p /opt/watch
 ---> Using cache
 ---> 63be987bef42
Step 4 : CMD inotifywait -mr --timefmt '%H:%M' --format '%T %w %e %f' /opt/watch
 ---> Using cache
 ---> d6a98d30c98e
Successfully built d6a98d30c98e
7a7q7vx0foxs1ucy9int21ve9
sleeping 1 of 10
sleeping 2 of 10
sleeping 3 of 10
sleeping 4 of 10
sleeping 5 of 10
sleeping 6 of 10
sleeping 7 of 10
sleeping 8 of 10
sleeping 9 of 10
sleeping 10 of 10
Setting up watches.  Beware: since -r was given, this may take a while!
Watches established.
11:03 /opt/watch/ CREATE 1.txt
11:03 /opt/watch/ MODIFY 1.txt
11:03 /opt/watch/ CREATE 2.txt
11:03 /opt/watch/ MODIFY 2.txt
11:03 /opt/watch/ CREATE 3.txt
11:03 /opt/watch/ MODIFY 3.txt
11:03 /opt/watch/ CREATE 4.txt
11:03 /opt/watch/ MODIFY 4.txt
11:03 /opt/watch/ CREATE 5.txt
11:03 /opt/watch/ MODIFY 5.txt
11:03 /opt/watch/ CREATE 6.txt
11:03 /opt/watch/ MODIFY 6.txt
11:03 /opt/watch/ CREATE 7.txt
11:03 /opt/watch/ MODIFY 7.txt
11:03 /opt/watch/ CREATE 8.txt
11:03 /opt/watch/ MODIFY 8.txt
11:03 /opt/watch/ CREATE 9.txt
11:03 /opt/watch/ MODIFY 9.txt
docker_swarm
```

**Comment by BretFisher** [comment]
💥

**Comment by tsabat** [comment]
You are the best @rheinwein! Thanks for this input.  The docs re: binding are few and far between.

@BretFisher you're cool too.

**Comment by tsabat** [comment]
@rheinwein bummer.  This seems not to work for me.  I still can't see the events in my container.  You said you can.  Can we determine what the differences are between your env and mine?

Here's what my bind looks like.  

image

and my docker version

image

and my Docker for Mac

image

@BretFisher can you get this project to build?

**Comment by rheinwein** [comment]
I ran this again this morning, and `bind-propagation` is set to `rprivate` even with the override. The Swarm example from this PR no longer works, but it looks like `bind-propagation` is working as (mostly) expected as I'm able to set it to `shared` but not to empty string.

**Pull request merged**